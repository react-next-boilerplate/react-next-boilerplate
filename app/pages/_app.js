import React from 'react';

import { Provider } from 'react-redux';

import Head from 'next/head';
import NextApp, { Container } from 'next/app';

import withReduxStore from 'utils/with-redux-store';

class Srr extends NextApp {
  render() {
    const { Component, pageProps, reduxStore } = this.props;
    return (
      <Container>
        <Head>
          <title>My page</title>
        </Head>

        <Provider store={reduxStore}>
          <Component {...pageProps} />
        </Provider>
      </Container>
    );
  }
}

export default withReduxStore(Srr);
