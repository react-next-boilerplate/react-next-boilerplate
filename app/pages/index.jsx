import React from 'react';

import styles from './styles.scss';

class Index extends React.PureComponent {
  render() {
    return (
      <div className={styles.root}>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis modi
        unde, sed nihil sapiente numquam ipsam earum rem blanditiis totam
        aperiam error, assumenda vel adipisci. Unde velit aliquid autem ratione.
      </div>
    );
  }
}

export default Index;
