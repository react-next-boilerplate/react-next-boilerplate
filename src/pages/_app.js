import React from 'react';

import { Provider } from 'react-redux';
import Head from 'next/head';
import App from 'next/app';

import withReduxStore from 'utils/with-redux-store';
import { appWithTranslation } from 'utils/with-i18next';

import 'fontsource-metropolis';
import '@typefaces-pack/typeface-inter';

class Srr extends App {
  render() {
    const { Component, pageProps, reduxStore } = this.props;
    return (
      <React.StrictMode>
        <Head>
          <title>React Next Boilerplate</title>
        </Head>

        <Provider store={reduxStore}>
          <Component {...pageProps} />
        </Provider>
      </React.StrictMode>
    );
  }
}

export default appWithTranslation(withReduxStore(Srr));
