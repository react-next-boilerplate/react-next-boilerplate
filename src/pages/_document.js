import React from 'react';
import Document, { Html, Head, Main, NextScript } from 'next/document';

import { Global, css } from '@emotion/react';

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const page = ctx.renderPage();
    // extract css to render in SSR
    const initialProps = await Document.getInitialProps(ctx);
    return { ...initialProps, ...page };
  }

  render() {
    return (
      <Html>
        <Global
          styles={css`
            ::selection {
              background-color: #26bd5a;
              color: #fff;
            }
            body,
            html {
              margin: 0;
              padding: 0;
              font-family: 'Inter';
            }
          `}
        />

        <Head>
          <link rel="apple-touch-icon" sizes="180x180" href="/static/favicon/apple-touch-icon.png" />
          <link rel="icon" type="image/png" sizes="32x32" href="/static/favicon/favicon-32x32.png" />
          <link rel="icon" type="image/png" sizes="16x16" href="/static/favicon/favicon-16x16.png" />
          <link rel="manifest" href="/static/favicon/site.webmanifest" />
          <link rel="mask-icon" href="/static/favicon/safari-pinned-tab.svg" color="#000000" />
          <link rel="shortcut icon" href="/static/favicon/favicon.png" />
          <meta name="msapplication-config" content="/static/favicon/browserconfig.xml" />
          <meta name="theme-color" content="#000" />
        </Head>

        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
