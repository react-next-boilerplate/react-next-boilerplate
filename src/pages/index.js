import React from 'react';
import PropTypes from 'prop-types';

import Home from 'containers/Home';
import { withTranslation } from 'utils/with-i18next';

export class IndexPage extends React.PureComponent {
  render() {
    const { t } = this.props;

    return <Home t={t} />;
  }
}

IndexPage.propTypes = {
  t: PropTypes.func,
};

IndexPage.getInitialProps = async () => ({
  namespacesRequired: ['common', 'banner', 'features'],
});

export default withTranslation('common')(IndexPage);
