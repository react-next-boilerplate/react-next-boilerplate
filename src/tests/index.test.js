import React from 'react';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'utils/configure-store';

import { IndexPage } from '../pages/index';

describe('<HomePage />', () => {
  let store;
  const spy = jest.fn();

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render and match the snapshot', () => {
    let props;

    act(() => {
      props = IndexPage.getInitialProps();
    });

    const { container } = render(
      <Provider store={store}>
        <IndexPage t={spy} {...props} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
