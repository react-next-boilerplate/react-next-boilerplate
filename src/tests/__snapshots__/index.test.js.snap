// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <header
    class="css-15etwxy-HeaderRoot ez4dfsd5"
  >
    <div
      class="css-kubhes-HeaderContainer ez4dfsd4"
    >
      <nav
        class="css-11njdmq-NavRoot ez4dfsd3"
      >
        <div
          class="css-1hbc6ct-ListItem ez4dfsd2"
        >
          <a
            class="css-15ani8u-ItemLink epoma0f0"
            href="/#features"
          >
            phrases.features
          </a>
          <a
            class="css-15ani8u-ItemLink epoma0f0"
            href="/#exampleGetApi"
          >
            phrases.apiExample
          </a>
        </div>
        <div
          class="css-13ln5v8-Space ez4dfsd0"
        />
        <div
          class="css-1at58d5-ListActions ez4dfsd1"
        >
          <select
            class="css-1j005nt-SelectRoot e1ut4ry90"
            id="languages"
            name="languages"
          >
            <option
              value="es"
            >
              languages.es
            </option>
            <option
              value="en"
            >
              languages.en
            </option>
          </select>
        </div>
      </nav>
    </div>
  </header>
  <main>
    <div
      class="css-12g2fob-BannerRoot eywty1z3"
    >
      <img
        alt="react-next-boilerplate"
        class="css-1gb4ms7-Logo eywty1z0"
        src="/static/images/reac-next-boilerplate-logo.svg"
      />
      <h1
        class="css-3j23gb-Title eywty1z2"
      >
        title
      </h1>
      <h2
        class="css-15g2rhr-SubTitle eywty1z1"
      >
        subTitle
      </h2>
    </div>
    <div
      class="css-1a7ifp2-Container e18s5a7r6"
      id="features"
    >
      <div
        class="css-2gab7u-FeaturesRoot e18s5a7r5"
      >
        <div
          class="css-4k3a82-FeaturesListContainer e18s5a7r3"
        >
          <div
            class="css-1iwp3y-FeaturesList e18s5a7r4"
          >
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                SEO
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.seo
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                redux-saga
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.reduxSaga
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                next-i18next
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.nextI18next
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                i18next
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.i18next
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                redux
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.redux
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                reselect
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.reselect
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                Immer
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.immer
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                @testing-library/react
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.testingLibraryReact
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                Typefaces
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.typefaces
              </p>
            </div>
            <div
              class="css-zinlcv-FeatureItem e18s5a7r2"
            >
              <h3
                class="css-k216p8-Title e18s5a7r1"
              >
                Emotion
              </h3>
              <p
                class="css-168i06-Content e18s5a7r0"
              >
                features.emotion
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-ycbew2-Container e1vuap0q6"
      id="exampleGetApi"
    >
      <button
        class="css-cms8ac-ButonGetApi e1vuap0q5"
      >
        phrases.getShowcases
      </button>
    </div>
  </main>
  <footer
    class="jsx-898115948 root"
  >
    <div
      class="jsx-898115948 container"
    >
      <div
        class="jsx-898115948 row"
      >
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/server-side-rendering"
          rel="noopener noreferrer"
          target="_blank"
        >
          SSR
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/static-exporting"
          rel="noopener noreferrer"
          target="_blank"
        >
          Static Exporting
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/css-in-js"
          rel="noopener noreferrer"
          target="_blank"
        >
          CSS-in-JS
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/progressive-web-apps"
          rel="noopener noreferrer"
          target="_blank"
        >
          PWA
        </a>
      </div>
    </div>
  </footer>
</div>
`;
